src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -415,32 +340,32 @@ private void StartWriting(byte[] buffer, int offset, int count, AsyncProtocolReq
                     }
 
                     int chunkBytes = Math.Min(count, _sslState.MaxDataSize);
-                    int encryptedBytes;
+                    byte[] rentedBuffer = ArrayPool<byte>.Shared.Rent(chunkBytes + FrameOverhead);
+                    byte[] outBuffer = rentedBuffer;
```

Comment:
outBuffer and rentedBuffer seem to always be the same.  Can we clean this up?

Reply:
A call below takes `outBuffer` by ref and may change it, and if it does we still need to be sure we're releasing the originally rented before to the pool, not whatever `outBuffer` was reassigned to by that ref.

Reply:
`outBuffer` is passed by `ref` so isn't necessarily the same as `rentedBuffer` after the call?

Reply:
Yeah, I missed the ref, thanks.

The fact that outBuffer is passed by ref is weird in and of itself.  Lower layers seem to do a null check and buffer size check, and allocate byte[] if the original buffer is null or too small.  This is almost certainly the wrong thing to do; we should just ensure that the originally allocated buffer is large enough.

That said, that's a problem for another day.

Reply:
Absolutely and before this change it was often the case, mostly because PinnableCache was fixed in size. This changes that but I didn't want to go through the layers and rip stuff out .... yet... as it touched all the PALs. I am trying to avoid massive sweeping changes in jumbo PR's :)

Reply:
That's good :)